### PR TITLE
Remove FishData

### PR DESCRIPTION
**Pull request by sovdeeth**
### Description

FishData was a bit of a black sheep in the family, where it was in charge of handling multiple classes instead of just one class with variants. It also didn't work at all. This PR removes it, replacing it with a supertype data. This means all the spawning is now part of the already-registered simple datas (or TropicalFishData), which all work. It maintains the comparability of fish, but means that you can't spawn a random fish with `spawn a fish`. Given that this was broken prior to this PR, I don't think it's much of an issue. It also follows the same behavior as `spawn a mob`.

---
**Target Minecraft Versions:** any 
**Requirements:** none 
**Related Issues:** #6707

### CODE REVIEW

File: src/test/skript/tests/regressions/6707-spawn-fish-types.sk [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+	spawn a tropical fish at spawn of world "world":
+		set {_e} to entity
+	assert {_e} is a tropical fish with "failed to spawn a tropical fish"
+	delete entity within {_e}
+
+	parse if running minecraft "1.19":
+		spawn a tadpole at spawn of world "world":
+			set {_e} to entity
+		assert {_e} is a tadpole with "failed to spawn a tadpole"
+		delete entity within {_e}
```
Comment (Fusezion) [inline src/test/skript/tests/regressions/6707-spawn-fish-types.sk:26]: should this be parse or `test "blah" if running minecraft 1.19`? or whatever the syntax was for version testing

Comment (sovdeeth) [reply src/test/skript/tests/regressions/6707-spawn-fish-types.sk:26]: That'd make it a different test, and I believe I'd still need to gate it behind a parse if. `tadpole` just doesn't exist prior to 1.19.

Comment (Fusezion) [reply src/test/skript/tests/regressions/6707-spawn-fish-types.sk:26]: taking ExprFreezeTicks test as an example `test "freeze time" when running minecraft "1.18":` pretty sure you don't

**Review by APickledWalrus — APPROVED** [review]

**Review by Moderocky — CHANGES_REQUESTED** [review]
Just reminding me to look at this later.

**Comment by sovdeeth** [comment]
> Just reminding me to look at this later.

look at it now, nerd

**Review by Moderocky — APPROVED** [review]

**Pull request merged**